x/internal/outline/access_key.go

```diff
@@ -25,15 +25,15 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport/shadowsocks"
 )
 
-type sessionConfig struct {
+type Prefix []byte
+
+type SessionConfig struct {
```

Comment:
This doesn't match our dynamic key config JSON. If we want to expose a format, we should use either the access key or the config we support.

Reply:
You are right, however `outline-connectivity` is referencing these fields. I think ideally the type should be:

```
type SessionConfig struct {
  // all private, no public fields
}

func (SessionConfig) String() string {
  // this method can be used by connectivity test to print the structured config object
}
```

Reply:
I would still like to keep this intermediate config private and only provide methods that take an access key and return objects.

That doesn't work for outline-connectivity, so let's revert outline-connectivity. It's ok if it duplicates code for now, we can figure that out later.

Reply:
We may want to provide some sort of shadowsocks-specific public library to parse Shadowsocks URLs, but let's do that later. It needs more discussion. The config area is a whole can of worms.